mets_adapter/mets_adapter/src/main/scala/uk/ac/wellcome/mets_adapter/services/MetsAdapterWorkerService.scala

```diff
@@ -56,6 +55,10 @@ class MetsAdapterWorkerService(
       className,
       source => {
         source
+          .map {
+            case (msg, NotificationMessage(body)) =>
+              (msg, fromJson[IngestUpdate](body).get)
```

Comment:
Just wondering if this is desirable behaviour in the case where it cannot parse the message.

We could do this to go through the error handling code:

```scala
.map {
   case (msg, NotificationMessage(body)) =>
    (msg, fromJson[IngestUpdate](body).toEither)
}
.via(catchErrors)
```

Reply:
Actually would need to a bit of modification so doesn't just accept a `Context` object..

Reply:
You're definitely right about the error handling. I'm not sure I understand the second bit about the `Context` object

Reply:
I mean  the `catchErrors` method  would have  to look something like this instead:

```scala
  def catchErrors[T, C] =
    Flow[(C, Result[T])]
      .map {
        case (ctx, result) =>
          result.left.map { err =>
            error(
              s"Error encountered processing SQS message. [Error]: ${err.getMessage} [Context]: $ctx",
              err)
          }
          (ctx, result)
      }
      .collect {
        case (ctx, Right(data)) => (ctx, data)
      }

```

Reply:
thanks, makes sense